### PR TITLE
1.5.7

### PR DESCRIPTION
**Pull request by isbm**
Checklist of done:

** DONE Use systemd (bsc#1024058)
** DONE Prevent /var/lib/pgsql/data (bsc#1024058)
** DONE Remove nag-screen every time shown
** DONE Update nag-screen to the current data (company name, years etc)
** DONE Add cleanup feature (remove pid). This should be transparent with the db-start function.
** DONE Take a snapshot of the current database configuration (include all files from /var/lib/pgsql/data)

### CODE REVIEW

File: src/smdba/postgresqlgate.py [hunk 1]
```diff
@@ -197,10 +197,14 @@ def check(self):
...
         elif not os.path.exists("/usr/bin/pg_basebackup"):
             msg = 'backup'
-
         if msg:
             raise GateException("Cannot find required %s component." % msg)
 
+        # Prevent running this tool within the PostgreSQL data directory
+        # See bsc#1024058 for details
+        if os.path.abspath(".") == self.config["pcnf_pg_data"]:
+            raise GateException("Please do not call SMDBA inside {0} directory.".format(os.path.abspath(".")))
```
Comment (mcalmer) [inline src/smdba/postgresqlgate.py]: What would happen if I am in /var/lib/pgsql/data/pg_log/ dir and call this ?

File: src/smdba/postgresqlgate.py [hunk 2]
```diff
@@ -188,19 +191,23 @@ def check(self):
...
             msg = 'backup'
-
         if msg:
             raise GateException("Cannot find required %s component." % msg)
 
+        # Prevent running this tool within the PostgreSQL data directory
+        # See bsc#1024058 for details
+        if os.path.abspath(".") == self.config["pcnf_pg_data"]:
+            raise GateException("Please do not call SMDBA inside the '{0}' directory.".format(os.path.abspath(".")))
+
```
Comment (mcalmer) [inline src/smdba/postgresqlgate.py]: Seems my comment was marked as outdated. So I repeat it:-)

I see problems with this exact compare. What happens if:
- self.config["pcnf_pg_data"] is sometimes configured with a final / and and sometimes without?
- what if I am in a subdir of self.config["pcnf_pg_data"] like /var/lib/pgsql/data/pg_log/ ?

I think we need to check if we are somewhere inside that dir.

Comment (isbm) [inline src/smdba/postgresqlgate.py]: @mcalmer good catch, BTW. :+1:

**Comment by isbm** [comment]
@mcalmer fixd

**Comment by mcalmer** [comment]
Should be good enough to catch a lot of possible customer failures :-)

**Pull request merged**